src/libs/Navigation/Navigation.js

```diff
@@ -54,27 +60,38 @@ function goBack(shouldOpenDrawer = true) {
  * @param {String} route
  */
 function navigate(route = ROUTES.HOME) {
-    if (route === ROUTES.HOME) {
-        if (isLoggedIn) {
-            openDrawer();
+    if (navigationRef.isReady()) {
```

Comment:
This a not a good idea. Due to this check calls to navigate will be ignored when navigationRef is not ready and the developer will not be able to know that navigate usages is wrong. 

Originally, this method should throw an error when `navigationref` is not ready and thus we know that we have called the `navigate` at the wrong place & we should refactor code to make that work.

Reply:
I don't think it's a bad idea per se (we explored it in the past IIRC), but maybe better to set something like a `initialReportID` that the navigation setup can use instead of this method here:

https://github.com/Expensify/App/blob/12200433c31f742f9aa7934c3d388dd8bb2dfa5e/src/libs/Navigation/AppNavigator/MainDrawerNavigator.js#L28-L34